File: backend/src/main/java/br/ufpr/sept/so2/modules/iam/application/ports/JwtTokenService.java
package br.ufpr.sept.so2.modules.iam.application.ports;

import br.ufpr.sept.so2.modules.iam.domain.Usuario;

import java.util.List;
import java.util.UUID;

public interface JwtTokenService {

    String emitAccessToken(Usuario usuario);

    String emitResetToken(Usuario usuario);

    AccessTokenClaims parseAccessToken(String token);

    ResetTokenClaims parseResetToken(String token);

    record AccessTokenClaims(
            UUID userId,
            List<String> authorities,
            boolean mustChangePassword,
            String jti
    ) {
    }

    record ResetTokenClaims(UUID userId, String jti) {
    }
}

File: backend/src/main/kotlin/br/ufpr/sept/so2/modules/iam/application/ports/JwtTokenService.kt
package br.ufpr.sept.so2.modules.iam.application.ports

import br.ufpr.sept.so2.modules.iam.domain.Usuario
import java.util.UUID

interface JwtTokenService {
    fun emitAccessToken(usuario: Usuario): String

    fun emitResetToken(usuario: Usuario): String

    fun parseAccessToken(token: String): AccessTokenClaims

    fun parseResetToken(token: String): ResetTokenClaims

    data class AccessTokenClaims(
        val userId: UUID,
        val authorities: List<String>,
        val mustChangePassword: Boolean,
        val jti: String,
    )

    data class ResetTokenClaims(
        val userId: UUID,
        val jti: String,
    )
}
